tests/foreman/api/test_audit.py

```diff
@@ -0,0 +1,175 @@
+
+        :expectedresults: Audit logs contain corresponding entries per each
+            create event
+
+        :BZ: 1426742
+
+        :CaseImportance: Critical
+        """
+        for entity_item in [
+            {'entity': entities.Architecture()},
+            {
+                'entity': entities.AuthSourceLDAP(),
+                'entity_type': 'auth_source',
+                'value_template': 'LDAP-{}'
+            },
+            {
+                'entity': entities.ComputeProfile(),
+                'entity_type': 'compute_profile'
+            },
+            {
+                'entity': entities.LibvirtComputeResource(),
+                'entity_type': 'compute_resource',
+                'value_template': '{} (Libvirt)'
+            },
+            {'entity': entities.ConfigGroup(), 'entity_type': 'config_group'},
+            {'entity': entities.Domain()},
+            {'entity': entities.Host()},
+            {'entity': entities.HostGroup()},
+            {'entity': entities.Image(
+                compute_resource=entities.LibvirtComputeResource().create())},
+            {'entity': entities.Location()},
+            {'entity': entities.Media(), 'entity_type': 'medium'},
+            {'entity': entities.Organization()},
+            {'entity': entities.OperatingSystem(), 'entity_type': 'os'},
+            {
+                'entity': entities.PartitionTable(),
+                'entity_type': 'partition_table',
+                'bz_id': 1492668
+            },
+            {'entity': entities.PuppetClass()},
+            {'entity': entities.Role()},
+            {'entity': entities.Subnet()},
+            {
+                'entity': entities.ProvisioningTemplate(),
+                'entity_type': 'template',
+                'bz_id': 1492696
+            },
+            {'entity': entities.User()},
+            {'entity': entities.UserGroup()},
+        ]:
+            bug_id = entity_item.get('bz_id', None)
+            if bug_id is not None and bz_bug_is_open(bug_id):
+                continue
+            created_entity = entity_item['entity'].create()
+            entity_type = entity_item.get(
+                'entity_type', created_entity.__class__.__name__.lower())
+            if entity_type == 'user':
+                entity_value = created_entity.login
+            else:
+                entity_value = created_entity.name
+            value_template = entity_item.get('value_template', None)
+            if value_template is not None:
+                entity_value = value_template.format(created_entity.name)
+            if entity_type == 'os':
+                entity_value += ' {}'. format(created_entity.major)
+            if entity_type == 'subnet':
+                entity_value += ' ({}/{})'. format(
+                    created_entity.network, created_entity.cidr
+                )
+            audit = entities.Audit().search(
+                query={'search': 'type={0}'.format(entity_type)})[0]
```

Comment:
are you sure that the first one, will be always the one that we added upper and  not an other one created by other test?

Reply:
Same concern. There's always a chance some kind of background task (e.g. from failed test) will interfere. I'd rather specify more specific search query

Reply:
I know the test are marked with run_in_one_thread, but it's still won't guarantee us nothing is running in the background.

Reply:
right, but, there is no other way. Believe me =)
There are no id or name parameters, so no way to make sure that it is correct element. More than 100 runs, no error, so far. If it ever fail - please let me know

Reply:
of course, we can make one more search inside of the list returned, but so far no necessity to do that

Reply:
That's one of options, another one is to use `filters=` arg. But that's very weird - neither auditable_id, nor even name or label are searchable, imo it's worth filling bz for that, since such issue makes those logs completely unmanageable.

Reply:
perhaps create a new  user and  create the entities by that user  and search by that user and entity_type 
what do you think?

Reply:
@ldjebran 
```
"error": {"message":"Field 'user_id' not recognized for searching!","class":"ScopedSearch::QueryNotSupported"}
```